### PR TITLE
bindinfo, planner: fix sql bind when sql has symbol list

### LINKED ISSUE #13871: sql bind will not be hit if the sql is similar as `select * from x limit 0, 1`
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

```mysql
create table t(a int, b int, INDEX ia (a), INDEX ib (b));
insert into t value(1, 1);

create global binding for select a, b, sleep(3) from t where a = 1 limit 0, 1 using select a, b, sleep(3) from t use index (ib) where a = 1 limit 0, 1;
```

Then run,
```mysql
select a, b, sleep(3) from t where a = 1 limit 0, 1;
```

2. What did you expect to see?
During the sleeping, show the execution plan in another client, but it still uses index `ia`
```mysql
mysql> show processlist;
+------+------+-----------+------+---------+------+-------+-----------------------------------------------------+
| Id   | User | Host      | db   | Command | Time | State | Info                                                |
+------+------+-----------+------+---------+------+-------+-----------------------------------------------------+
|    1 | root | 127.0.0.1 | test | Query   |    0 | 2     | select a, b, sleep(3) from t where a = 1 limit 0, 1 |
|    2 | root | 127.0.0.1 | test | Query   |    0 | 2     | show processlist                                    |
+------+------+-----------+------+---------+------+-------+-----------------------------------------------------+
2 rows in set (0.00 sec)

mysql> explain for connection 1;
+--------------------------+-------+-----------+---------------------------------------------------------------+
| id                       | count | task      | operator info                                                 |
+--------------------------+-------+-----------+---------------------------------------------------------------+
| Projection_7             | 1.00  | root      | Column#1, Column#2, sleep(3)                                  |
| └─IndexLookUp_18         | 1.00  | root      | limit embedded(offset:0, count:1)                             |
|   ├─Limit_17             | 1.00  | cop[tikv] | offset:0, count:1                                             |
|   │ └─IndexScan_15       | 1.00  | cop[tikv] | table:t, index:a, range:[1,1], keep order:false, stats:pseudo |
|   └─TableScan_16         | 1.00  | cop[tikv] | table:t, keep order:false, stats:pseudo                       |
+--------------------------+-------+-----------+---------------------------------------------------------------+
5 rows in set (0.00 sec)
```

3. What did you see instead?

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

### PR DESCRIPTION
**Pull request by SunRunAway**
### What problem does this PR solve? 

Fix #13871, related PR https://github.com/pingcap/parser/pull/656

### What is changed and how it works?

Introduce `parser.DigestNormalized` to avoiding twice normalized.

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

Code changes

 - None

Side effects

 - None

Related changes

 - None

Release note

 - fix that SQL bind cannot be hit when a SQL has symbol list

### CODE REVIEW

File: bindinfo/bind_test.go [hunk 1]
```diff
@@ -365,6 +365,43 @@ func (s *testSuite) TestExplain(c *C) {
...
+func (s *testSuite) TestBindingSymbolList(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	s.cleanBindingEnv(tk)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(a int, b int, INDEX ia (a), INDEX ib (b));")
+	tk.MustExec("insert into t value(1, 1);")
+
+	// before binding
+	tk.MustQuery("select a, b, sleep(1) from t where a = 3 limit 1, 100")
```
Comment (lamxTyler) [inline bindinfo/bind_test.go]: Why `sleep(1)`? Won't it slow down the test?

Comment (SunRunAway) [reply bindinfo/bind_test.go]: Removed.

File: go.sum [hunk 2]
```diff
@@ -235,6 +239,8 @@ github.com/shurcooL/vfsgen v0.0.0-20181020040650-a97a25d856ca/go.mod h1:TrYk7fJV
 github.com/sirupsen/logrus v1.0.5/go.mod h1:pMByvHTf9Beacp5x1UXfOR9xyW/9antXMhjMPG0dEzc=
 github.com/sirupsen/logrus v1.2.0 h1:juTguoYk5qI21pwyTXY3B3Y5cOTH3ZUyZCg1v/mihuo=
 github.com/sirupsen/logrus v1.2.0/go.mod h1:LxeOpSwHxABJmUn/MG1IvRgCAasNZTLOkJPxbbu5VWo=
+github.com/sirupsen/logrus v1.4.2 h1:SPIRibHv4MatM3XXNO2BJeFLZwZ2LvZgfQ5+UNI2im4=
+github.com/sirupsen/logrus v1.4.2/go.mod h1:tLMulIdttU9McNUspp0xgXVQah82FyeX6MwdIuYE2rE=
```
Comment (lamxTyler) [inline go.sum]: It is better to not introduce these kind of unrelated changes, and `make tidy` before push.

Comment (SunRunAway) [reply go.sum]: Update.

**Pull request merged**